src/app/shared/items/item-type-decorator.ts

```diff
@@ -0,0 +1,60 @@
+import { hasNoValue, hasValue } from '../empty.util';
+import { MetadataRepresentationType } from '../../core/shared/metadata-representation/metadata-representation.model';
+import { VIEW_MODE_ELEMENT } from '../../+item-page/simple/related-items/related-items-component';
+
+export const DEFAULT_ITEM_TYPE = 'Default';
+export const DEFAULT_VIEW_MODE = VIEW_MODE_ELEMENT;
+export const NO_REPRESENTATION_TYPE = MetadataRepresentationType.None;
+export const DEFAULT_REPRESENTATION_TYPE = MetadataRepresentationType.PlainText;
+
+const map = new Map();
+
+/**
+ * Decorator used for rendering simple item pages by type and viewMode (and optionally a representationType)
+ * @param type
+ * @param viewMode
+ * @param representationType
+ */
+export function rendersItemType(type: string, viewMode: string, representationType?: MetadataRepresentationType) {
```

Comment:
Shouldn't this decorator be capitalized?
I would suggest having a different name, perhaps: Item Type Rendering/View Mode

Reply:
I couldn't find any rules in the [style guide](https://angular.io/guide/styleguide) about capitalizing decorators. The decorators inside Angular seem to be capitalized, but then the ones of [the third party libs we use, aren't](https://github.com/DSpace/dspace-angular/blob/da1a97bd8cf49365e080884c2f622a6fb98106cc/src/app/core/cache/models/normalized-dspace-object.model.ts#L30), neither are [the other custom decorators we wrote before](https://github.com/DSpace/dspace-angular/blob/da1a97bd8cf49365e080884c2f622a6fb98106cc/src/app/shared/object-grid/item-grid-element/item-grid-element.component.ts#L14). So an argument could be made _not_ to capitalize third party decorators, to make it clear they aren't part the framework. 

In any case if we decide to change the way we approach that, I'd do it in a separate PR that fixes it for all our custom decorators at once, not just the ones in this PR.

I also prefer names that read like a sentence. if I see `@rendersItemType(Person, Grid)` I read that as "renders Items (of the) type Person, (in a) Grid" whereas `@ItemTypeRendering(Person, Grid)` requires some more deciphering

Reply:
You are right, that is nothing in the style guide that specifically says how decorators should be writen. But in all the presented examples, they are capitalized.
I also found this: https://toddmotto.com/angular-decorators#creating-a-decorator
If others have the same opinion, perhaps is best to refactor the code in another pull request.

Regarding the name, I think is a subjective matter. To me ` @ItemTypeViewMode(Person, Grid)`
works best than `@rendersItemType(Person, Grid)`. But it's personal option.